vantage6-algorithm-store/vantage6/algorithm/store/resource/algorithm.py

```diff
@@ -173,11 +173,16 @@ def _get_image_digest(self, image_name: str) -> tuple[str, str]:
                     registry_password = reg.get("password")
                     break
             if registry_user and registry_password:
+                log.info("Retrying to get digest with authentication...")
                 digest = get_digest(
                     full_image=image_name,
                     registry_username=registry_user,
                     registry_password=registry_password,
                 )
+                if digest:
+                    log.info("Digest obtained succesfully!")
+                else:
+                    log.error("Failed to get digest with authentication!")
```

Comment:
I guess we should also log it when no digest could be obtained and there are no credentials available. Careful with accepting this suggestion as the indentation might be wrong:
```suggestion
                    log.error("Failed to get digest with authentication!")
                else:
                    log.error("Failed to get digest")
```

Also is there an particular reason you don't start out with checking if there are credentials? Then we don't have to do the request twice?